api/cloud/oc_cloud_manager.c

```diff
@@ -600,32 +631,37 @@ cloud_manager_login_handler(oc_client_response_t *data)
         refresh_token_expires_in_ms(ctx->store.expires_in *
                                     MILLISECONDS_PER_SECOND));
     }
-    goto finish;
-  }
-  if (ret == CLOUD_ERROR_UNAUTHORIZED) {
-    if (handleUnauthorizedByRefresh) {
-      oc_reset_delayed_callback(ctx, cloud_manager_refresh_token_async,
-                                g_retry_timeout[0]);
-    }
-    goto finish;
-  }
-
-  if ((ctx->store.status & OC_CLOUD_REGISTERED) == 0) {
-    goto finish;
+    oc_reset_delayed_callback(ctx, cloud_manager_callback_handler_async, 0);
+    return;
   }
-  if (!is_retry_over(ctx)) {
-    retry = true;
-  } else {
-    oc_reset_delayed_callback(ctx, cloud_manager_reconnect_async, 0);
+  ctx->store.status |= OC_CLOUD_FAILURE;
+  if (err == CLOUD_ERROR_CONNECT &&
+      cloud_schedule_retry(ctx, OC_CLOUD_ACTION_LOGIN,
```

Comment:
in case `err == CLOUD_ERROR_CONNECT` but `cloud_schedule_retry` fails we will get here, but `cps` is set to `cps_ok` in `_login_handler`, but it should be `OC_CPS_FAILED` in this case.

so maybe something like this:
```C
if (err == CLOUD_ERROR_CONNECT) {
  if (cloud_schedule_retry(ctx, OC_CLOUD_ACTION_LOGIN, cloud_manager_login_async, data->code) {
    ...
    return;
  }
  cps = OC_CPS_FAILED;
}

```

Reply:
fixed